### PR TITLE
NXP-26348: improve versioning

### PR DESCRIPTION
**Pull request by troger**

### CODE REVIEW

File: nuxeo-wopi/src/main/java/org/nuxeo/wopi/jaxrs/FilesEndpoint.java [hunk 1]
```diff
@@ -293,9 +293,9 @@ protected Object lock(String lock) {
             logCondition(() -> LOCK + " header is equal to current WOPI lock"); // NOSONAR
             // refresh lock
             LockHelper.refreshLock(fileId);
-            String versionLabel = doc.getVersionLabel();
-            response.addHeader(ITEM_VERSION, versionLabel);
-            logResponse(OPERATION_LOCK, OK.getStatusCode(), ITEM_VERSION, versionLabel);
+            String itemVersion = blob.getDigest();
+            response.addHeader(ITEM_VERSION, itemVersion);
+            logResponse(OPERATION_LOCK, OK.getStatusCode(), ITEM_VERSION, itemVersion);
```
Comment (efge) [inline nuxeo-wopi/src/main/java/org/nuxeo/wopi/jaxrs/FilesEndpoint.java]: Maybe factor this in a method as it's done 4 times in this file.

**Comment by efge** [comment]
Another option (maybe) is to use the change token. It's specific to the document and not the blob but may be enough for your use case. Although it changes for *any* doc modification so it may be too much.

**Comment by troger** [comment]
Do we have any issue using the blob digest here? It's finally the more specific to the blob, so it matches what's expected by Office Online.

**Comment by ataillefer** [comment]
I totally agree with @troger, we want a blob-related value, as required by Microsoft: "This value must change when the file changes, and version values must never repeat for a given file."

**Comment by ataillefer** [comment]
:+1:

**Comment by ataillefer** [comment]
:+1:

**Pull request merged**